src/ClassConfigFactory.php

```diff
@@ -64,29 +64,48 @@ private function getMethodConfigs(ReflectionClass $class): array
     {
         $methods = [];
         foreach ($class->getMethods() as $method) {
-            $methods[$method->getName()] = $this->getMethodConfig($method);
+            $methods[$method->getName()] = $this->getMethodConfig($class, $method);
         }
 
         return $methods;
     }
 
     /**
-     * Gets single method config for individual method reflection.
+     * Gets single method config for individual class / method reflection pair.
      *
+     * @param ReflectionClass $class Reflection of a class.
      * @param ReflectionMethod $method Reflection of a method.
      *
      * @return MethodConfig Single method config.
      */
-    private function getMethodConfig(ReflectionMethod $method): MethodConfig
+    private function getMethodConfig(ReflectionClass $class, ReflectionMethod $method): MethodConfig
     {
         return new MethodConfig(
-            modifiers: Reflection::getModifierNames($method->getModifiers()),
+            modifiers: $this->getMethodModifiers($class, $method),
             name: $method->getName(),
             parameters: $this->getMethodParameterConfigs($method),
             returnType: $this->getMethodReturnTypeConfig($method),
         );
     }
 
+    /**
+     * Gets the set of method modifiers for a given class / method reflection pair
+     *
+     * @param ReflectionClass $class Reflection of a class.
+     * @param ReflectionMethod $method Reflection of a method.
+     *
+     * @return string[] List of method modifiers.
+     */
+    private function getMethodModifiers(ReflectionClass $class, ReflectionMethod $method): array
+    {
+        $modifiers = Reflection::getModifierNames($method->getModifiers());
+        if (!$class->isInterface()) {
+            return $modifiers;
+        }
+
+        return array_values(array_filter($modifiers, fn (string $modifier) => $modifier !== 'abstract'));
```

Comment:
What's the logic to have `abstract` modifier in interface declaration?

Reply:
Do you mean in interface method declaration? It's vice versa. It was there before, because it's incorrectly included when getting modifiers through `Reflection` https://github.com/yiisoft./proxy/blob/401ccee0fc64a37aa9331e9f9966f79d62d04b1b/src/ClassConfigFactory.php#L47. Now `abstract` modifier is removed from interface methods.